### ISSUE TITLE
Allure-results does not exists - cant change the result path

### ISSUE THREAD

**Issue by semihural** [allure-framework/allure2#1133]
I am using Allure 2.28.1 plugin for Jenkins via declarative pipeline as below. But I could not able to read allure reports. So reports seems generated below` ~/workspace/selenium_tests@2/ApiTestAutomation/ApiHubTestAutomation/target` but I could not able to change the path in script.

How can I change the allure path to read the reports from below path ? Or even if I use dir and specify the path, why allure ignore this path to generate reports ?

```
ubuntu@play-0:~/workspace/selenium_tests@2/ApiTestAutomation/ApiHubTestAutomation/target$ ls                                                                                            
allure-results  generated-test-sources  maven-status  surefire  surefire-reports  test-classes

```
```
[selenium_tests] $ /home/ubuntu/tools/ru.yandex.qatools.allure.jenkins.tools.AllureCommandlineInstallation/Allure_2.13.5/bin/allure generate -c -o /home/ubuntu/workspace/selenium_tests/allure-report
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
allure-results does not exists
Report successfully generated to /home/ubuntu/workspace/selenium_tests/allure-report
Allure report was successfully generated.
Creating artifact for the build.
Artifact was added to the build.
```

Here is what I tried

```
        steps {
            script {
                dir ('~/workspace/selenium_tests@2/ApiTestAutomation/ApiHubTestAutomation') {
                    allure([
                            includeProperties: false,
                            jdk: '',
                            properties: [],
                            reportBuildPolicy: 'ALWAYS',
                            results: [[path: 'target/allure-results']]
                    ])
                }
            }
        }
    }
```

**Comment by nikhilmymail101** [comment]
@semihural 
can you share more console logs and maven pom.xml file???

**Comment by semihural** [comment]
@nikhilmymail101 
Here is the pom.xml and output of console logs.

logs.
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M4:test (default-test) on project xxx: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/ubuntu/workspace/selenium_tests@2/xxx/xxx/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :OneApiHubTestAutomation
[Pipeline] }
ERROR: script returned exit code 1
[Pipeline] // catchError
[Pipeline] }
[Pipeline] // dir
[Pipeline] }
[Pipeline] // script
Post stage
[Pipeline] script
[Pipeline] {
[Pipeline] publishHTML
[htmlpublisher] Archiving HTML reports...
[htmlpublisher] Archiving at BUILD level /home/ubuntu/workspace/selenium_tests@2/OneApiTestAutomation/OneApiHubTestAutomation/reports/oneapihub to /var/lib/jenkins/master/jobs/selenium_tests/builds/26/htmlreports/Selenium-HTML-Report
[Pipeline] }
[Pipeline] // script
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
$ docker stop --time=1 b405d6052f896f8d3ca6dea15dab49825589ce30336ba2cbe424e7d7c9d495ad
$ docker rm -f b405d6052f896f8d3ca6dea15dab49825589ce30336ba2cbe424e7d7c9d495ad
[Pipeline] // withDockerContainer
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Allure Reports)
[Pipeline] script
[Pipeline] {
[Pipeline] allure
[selenium_tests] $ /home/ubuntu/tools/ru.yandex.qatools.allure.jenkins.tools.AllureCommandlineInstallation/Allure_2.13.5/bin/allure generate -c -o /home/ubuntu/workspace/selenium_tests/allure-report
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
allure-results does not exists
Report successfully generated to /home/ubuntu/workspace/selenium_tests/allure-report
Allure report was successfully generated.
Creating artifact for the build.
Artifact was added to the build.
```

Here is pom.xml

```
<project xmlns="http://maven.apache.org/POM/4.0.0"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<groupId>com.oneapi.ta</groupId>
	<artifactId>xxxx</artifactId>
	<version>0.0.1-SNAPSHOT</version>
	<packaging>pom</packaging>
	<modules>
		<module>xxx</module>
		<module>xxx</module>
		<module>xxxx</module>
	</modules>
	<properties>
		<java.version>13</java.version>
		<cucumber.version>5.5.0</cucumber.version>
		<cucumber.provider>io.cucumber</cucumber.provider>
		<extentreports.version>4.1.3</extentreports.version>
		<testng.version>7.1.0</testng.version>
		<log4j.version>2.13.1</log4j.version>
		<aspectj.version>1.8.10</aspectj.version>
	</properties>
	<build>
		<plugins>
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-surefire-plugin</artifactId>
				<version>3.0.0-M4</version>
				<configuration>
					<argLine>
						-javaagent:"${settings.localRepository}/org/aspectj/aspectjweaver/${aspectj.version}/aspectjweaver-${aspectj.version}.jar"
					</argLine>
					<systemPropertyVariables>
						<allure.results.directory>${project.build.directory}/allure-results</allure.results.directory>
					</systemPropertyVariables>
				</configuration>
				<dependencies>
					<dependency>
						<groupId>org.aspectj</groupId>
						<artifactId>aspectjweaver</artifactId>
						<version>${aspectj.version}</version>
					</dependency>
				</dependencies>
			</plugin>
			<plugin>
				<groupId>io.qameta.allure</groupId>
				<artifactId>allure-maven</artifactId>
				<version>2.9</version>
				<configuration>
					<reportVersion>2.7.0</reportVersion>
					<allureDownloadUrl>https://github.com/allure-framework/allure2/releases/download/2.7.0/allure-2.7.0.zip</allureDownloadUrl>
				</configuration>
			</plugin>
		</plugins>
	</build>
	<dependencies>
	
			<!-- https://mvnrepository.com/artifact/io.qameta.allure/allure-cucumber5-jvm -->
		<dependency>
			<groupId>io.qameta.allure</groupId>
			<artifactId>allure-cucumber5-jvm</artifactId>
			<version>2.13.5</version>
		</dependency>
	
		<!-- https://mvnrepository.com/artifact/junit/junit -->
		<dependency>
			<groupId>junit</groupId>
			<artifactId>junit</artifactId>
			<version>4.13</version>
			<scope>test</scope>
		</dependency>

		<!-- https://mvnrepository.com/artifact/io.cucumber/cucumber-core -->
		<dependency>
			<groupId>io.cucumber</groupId>
			<artifactId>cucumber-core</artifactId>
			<version>${cucumber.version}</version>
		</dependency>


		<!-- https://mvnrepository.com/artifact/io.cucumber/cucumber-html -->
		<dependency>
			<groupId>io.cucumber</groupId>
			<artifactId>cucumber-html</artifactId>
			<version>0.2.7</version>
		</dependency>

		<!-- https://mvnrepository.com/artifact/io.cucumber/cucumber-java -->
		<dependency>
			<groupId>io.cucumber</groupId>
			<artifactId>cucumber-java</artifactId>
			<version>${cucumber.version}</version>
		</dependency>

		<!-- https://mvnrepository.com/artifact/io.cucumber/cucumber-junit -->
		<dependency>
			<groupId>io.cucumber</groupId>
			<artifactId>cucumber-junit</artifactId>
			<version>6.6.0</version>
			<scope>test</scope>
		</dependency>


		<!-- https://mvnrepository.com/artifact/io.cucumber/gherkin -->
<!-- 		<dependency> -->
<!-- 			<groupId>io.cucumber</groupId> -->
<!-- 			<artifactId>gherkin</artifactId> -->
<!-- 			<version>15.0.2</version> -->
<!-- 		</dependency> -->

		<!-- https://mvnrepository.com/artifact/io.cucumber/cucumber-jvm-deps -->
		<dependency>
			<groupId>io.cucumber</groupId>
			<artifactId>cucumber-jvm-deps</artifactId>
			<version>1.0.6</version>
			<scope>provided</scope>
		</dependency>

		<!-- https://mvnrepository.com/artifact/io.cucumber/messages -->
		<dependency>
			<groupId>io.cucumber</groupId>
			<artifactId>messages</artifactId>
			<version>10.0.3</version>
		</dependency>

		<!-- https://mvnrepository.com/artifact/net.masterthought/cucumber-reporting -->
		<dependency>
			<groupId>net.masterthought</groupId>
			<artifactId>cucumber-reporting</artifactId>
			<version>5.1.1</version>
		</dependency>


		<!-- https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium-java -->
		<dependency>
			<groupId>org.seleniumhq.selenium</groupId>
			<artifactId>selenium-java</artifactId>
			<version>3.141.59</version>
		</dependency>

		<!-- https://mvnrepository.com/artifact/com.aventstack/extentreports -->
		<dependency>
			<groupId>com.aventstack</groupId>
			<artifactId>extentreports</artifactId>
			<version>${extentreports.version}</version>
		</dependency>


		<!-- https://mvnrepository.com/artifact/commons-io/commons-io -->
		<dependency>
			<groupId>commons-io</groupId>
			<artifactId>commons-io</artifactId>
			<version>2.6</version>
		</dependency>

		<!-- https://mvnrepository.com/artifact/io.cucumber/cucumber-testng -->
		<dependency>
			<groupId>io.cucumber</groupId>
			<artifactId>cucumber-testng</artifactId>
			<version>${cucumber.version}</version>
		</dependency>


		<!-- https://mvnrepository.com/artifact/org.testng/testng -->
		<dependency>
			<groupId>org.testng</groupId>
			<artifactId>testng</artifactId>
			<version>${testng.version}</version>
			<scope>compile</scope>
		</dependency>

		<dependency>
			<groupId>org.apache.logging.log4j</groupId>
			<artifactId>log4j-api</artifactId>
			<version>${log4j.version}</version>
		</dependency>
		
		<dependency>
			<groupId>org.apache.logging.log4j</groupId>
			<artifactId>log4j-core</artifactId>
			<version>${log4j.version}</version>
		</dependency>
		
		<dependency>
			<groupId>io.github.bonigarcia</groupId>
			<artifactId>webdrivermanager</artifactId>
			<version>3.8.1</version>
		</dependency>
		
		<dependency>
			<groupId>com.sun.mail</groupId>
			<artifactId>javax.mail</artifactId>
			<version>1.6.0</version>
		</dependency>

		<dependency>
			<groupId>org.slf4j</groupId>
			<artifactId>slf4j-api</artifactId>
			<version>1.7.30</version>
		</dependency>
		
		<dependency>
		    <groupId>javax.xml.bind</groupId>
		    <artifactId>jaxb-api</artifactId>
		    <version>2.3.1</version>
		</dependency>
		
		<!-- https://mvnrepository.com/artifact/com.auth0/java-jwt -->
		<dependency>
		    <groupId>com.auth0</groupId>
		    <artifactId>java-jwt</artifactId>
		    <version>3.10.3</version>
		</dependency>
	</dependencies>


</project>
```

**Comment by baev** [comment]
did you specified allure-results as `target/allure-results` in Jenkins pipeline?

**Comment by semihural** [comment]
@baev ,
I am using declarative pipeline thus there is no any section in job's configuration such as post-build function. But here is my script.

So the issue is, reports (json files and attachments) generated under below path whereas while this stage is running allure's output is `Allure_2.13.5/bin/allure generate -c -o /home/ubuntu/workspace/selenium_tests/allure-report` where there is also allure-report but I tried to give both path but both of the reports display Na.

ubuntu@play-0:~/workspace/selenium_tests@2/OneApiTestAutomation/OneApiHubTestAutomation/target$ ls                                                                                            
allure-results  generated-test-sources  maven-status  surefire  surefire-reports  test-classes

```
ubuntu@play-0:~/workspace/selenium_tests/allure-report$ ls                                                                                                                                    
app.js  data  export  favicon.ico  history  index.html  plugins  styles.css  widgets
```

```
        stage('Allure Reports') {
            when {
                beforeAgent true
                allOf {
                    expression { params.projectName == 'OneApiHubTestAutomation'}
                    expression { params.BRANCH == "development" || params.BRANCH == "stage" || params.BRANCH == "master"   }
                    expression { params.RUN_SELENIUM_TEST == true}
                }
            }
            steps {
                script {
                    allure([
                        includeProperties: false,
                        jdk: '',
                        properties: [],
                        reportBuildPolicy: 'ALWAYS',
                        results: [[path: 'target/allure-results']]
                    ])
                }
            }
        } 
```

**Comment by nikhilmymail101** [comment]
@semihural 
1.  Use allure.results.directory in properties section of pom.xml file instead of maven surefire plugin.
2. Use same version of cucumber for all cucumber libraries.



You can use maven site plugin to generate the report. The report will be generated but you won't be able to see it on report section of build.
To see the report in jenkins UI, you need to use post build action allure command-line.

below plugin is working for me...

<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-site-plugin</artifactId>
				<version>${maven-site-plugin}</version>
				<executions>
					<execution>
						<id>execution</id>
						<goals>
							<goal>site</goal>
						</goals>
						<configuration>
							<testFailureIgnore>true</testFailureIgnore>
							<argLine>
								–javaagent:${settings.localRepository}/org/aspectj/aspectjweaver/${aspectj.version}/aspectjweaver-${aspectj.version}.jar
								-Dcucumber.options="--plugin
								io.qameta.allure.cucumber5jvm.AllureCucumber5Jvm"
							</argLine>
						</configuration>
					</execution>
				</executions>
				<dependencies>
					<dependency>
						<groupId>org.aspectj</groupId>
						<artifactId>aspectjweaver</artifactId>
						<version>${aspectj.version}</version>
					</dependency>
				</dependencies>
			</plugin>
			<plugin>
				<groupId>org.eclipse.jetty</groupId>
				<artifactId>jetty-maven-plugin</artifactId>
				<version>9.2.10.v20150310</version>
				<configuration>
					<webAppSourceDirectory>
						${project.build.directory}/site/allure-maven-plugin
					</webAppSourceDirectory>
					<stopKey>stop</stopKey>
					<stopPort>1234</stopPort>
				</configuration>
			</plugin>
		</plugins>
	</build>
	<reporting>
		<excludeDefaults>true</excludeDefaults>
		<plugins>
			<plugin>
				<groupId>io.qameta.allure</groupId>
				<artifactId>allure-maven</artifactId>
				<version>2.10.0</version>
				<configuration>
					<reportDirectory>${basedir}/AllureReports</reportDirectory>
					<reportVersion>2.8.1</reportVersion>
				</configuration>
			</plugin>
		</plugins>
	</reporting>

**Comment by semihural** [comment]
@nikhilmymail101 ,

I will try asap and let you know.. So within these changes, no need to change the allure path ? Cause I could not find anyway to change **-o <path>** while allure is running

**Comment by fescobar** [comment]
You have to specify the WORKSPACE directory, did you try this?:
```
            steps {
                script {
                    allure([
                        includeProperties: false,
                        jdk: '',
                        properties: [],
                        reportBuildPolicy: 'ALWAYS',
                        results: [[path: '$WORKSPACE/target/allure-results']]
                    ])
                }
            }

```

**Comment by semihural** [comment]
@fescobar , Yes I tried to specify absolute path but it did not work for me. Allure somehow ignore this results path.

**Comment by fescobar** [comment]
@semihural that is not the absolute path, you have to check the workspace in your pipeline, take a screenshot of your workspace in the pipeline showing the files, please.

**Comment by nikhilmymail101** [comment]
@semihural 
you have to specify allure.results.directory on top under properties section.

**Comment by semihural** [comment]
@fescobar ,

As I mentioned above, reports where json and attachments files are under below **path.:home/ubuntu/workspace/selenium_tests@2/OneApiTestAutomation/OneApiHubTestAutomation/target**

```
ubuntu@play-0:home/ubuntu/workspace/selenium_tests@2/OneApiTestAutomation/OneApiHubTestAutomation/target$ ls                                                                                            
allure-results  generated-test-sources  maven-status  surefire  surefire-reports  test-classes
```

```
ubuntu@play-0:home/ubuntu/workspace/selenium_tests/allure-report$ ls                                                                                                                                    
app.js  data  export  favicon.ico  history  index.html  plugins  styles.css  widgets
```

**Comment by semihural** [comment]
@nikhilmymail101 Could you please share how it looks like ?Shall I only change the results line?

**Comment by nikhilmymail101** [comment]
@semihural 
I am using something like this one.....
<properties>
		<maven.compiler.source>1.8</maven.compiler.source>
		<maven.compiler.target>1.8</maven.compiler.target>
		<aspectj.version>1.8.10</aspectj.version>
		<maven-resource-plugin.version>3.1.0</maven-resource-plugin.version>
		<maven-site.version>3.4</maven-site.version>
		<xml-apis.version>1.4.01</xml-apis.version>
		<allure.version>1.5.0</allure.version>
		<allure-report.version>1.5.0</allure-report.version>
		<allure-maven.version>2.6</allure-maven.version>
		<allure.results.directory>${basedir}/allure-results</allure.results.directory>
		<maven.build.timestamp.format>yyyy_MM_dd_HH_mm_ss</maven.build.timestamp.format>
		<timestamp>${maven.build.timestamp}</timestamp>
		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
		<surefire.rerunFailingTestsCount>2</surefire.rerunFailingTestsCount>
		<selenium.version>3.141.59</selenium.version>
		<poi.version>4.1.2</poi.version>
		<xmlbeans>3.1.0</xmlbeans>
		<commons.io>2.7</commons.io>
		<commons.csv>1.8</commons.csv>
		<cucumber.version>5.6.0</cucumber.version>
		<rest.assured>4.3.1</rest.assured>
		<testng>7.3.0</testng>
		<cucumber-reporting>5.3.1</cucumber-reporting>
		<maven-surefire-plugin>2.22.2</maven-surefire-plugin>
		<maven-compiler-plugin>3.8.1</maven-compiler-plugin>
		<maven-site-plugin>3.9.1</maven-site-plugin>
	</properties>

**Comment by semihural** [comment]
@fescobar 

Here is the screenshot and the script that ı tried as you mentioned.

```
        stage('Allure Reports') {
            when {
                beforeAgent true
                allOf {
                    expression { params.projectName == 'OneApiHubTestAutomation'}
                    expression { params.BRANCH == "development" }
                    expression { params.RUN_SELENIUM_TEST == true}
                }
            }
            steps {
                script {
                    allure([
                        includeProperties: false,
                        jdk: '',
                        properties: [],
                        reportBuildPolicy: 'ALWAYS',
                        results: [[path: "${env.WORKSPACE}/target/allure-results"]]
                    ])
                }
            }
        } 
```

![image](https://user-images.githubusercontent.com/38153463/93126820-877ffb80-f6d5-11ea-9675-6fefdfc07de7.png)

**Comment by fescobar** [comment]
@semihural I'm not talking about that. Enter the pipeline from the Jenkins UI, click in the last build and you will see an option named `workspace`, in there you will see files (screenshot of that). Also, try this `${WORKSPACE}` or print the directory content of WORKSPACE var `ls -a ${WORKSPACE}`

**Comment by semihural** [comment]
@fescobar 

Here is the screenshot.

![image](https://user-images.githubusercontent.com/38153463/93167604-a01a0100-f729-11ea-8436-5dd78f34676a.png)

**Comment by semihural** [comment]
So the issue is, even if ı tried to change the allure path somehow it always add this path(/home/ubuntu/workspace/selenium_tests/) front of the path that I want. Here is how it looks like

```
[selenium_tests] $ /home/ubuntu/tools/ru.yandex.qatools.allure.jenkins.tools.AllureCommandlineInstallation/Allure_2.13.5/bin/allure generate -c -o /home/ubuntu/workspace/selenium_tests/~/workspace/selenium_tests@2/OneApiTestAutomation/OneApiHubTestAutomation/target
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
allure-results does not exists
Report successfully generated to /home/ubuntu/workspace/selenium_tests/~/workspace/selenium_tests@2/OneApiTestAutomation/OneApiHubTestAutomation/target
```

**Comment by fescobar** [comment]
@semihural you don't have to put the absolute path because you don't know what is the name of the workspace. Sometimes is `selenium_tests` and sometimes is `selenium_tests@2` or it could be whatever, for that reason I'm telling you, you have to use the env variable $WORKSPACE. It's working for me. Also if your directory includes `@` is because you are running in parallel, you have to have that in mind.

**Comment by semihural** [comment]
I tried with below as well but it did not work for me. 


```
        stage('Allure Reports') {
            when {
                beforeAgent true
                allOf {
                    expression { params.projectName == 'OneApiHubTestAutomation'}
                    expression { params.BRANCH == "development" }
                    expression { params.RUN_SELENIUM_TEST == true}
                }
            }
            steps {
                script {
                    allure([
                        includeProperties: false,
                        jdk: '',
                        properties: [],
                        reportBuildPolicy: 'ALWAYS',
                        results: [[path: "${env.WORKSPACE}/target/allure-results"]]
                    ])
                }
            }
        } 
```

**Comment by fescobar** [comment]
@semihural I've found the solution. You need to specify the workspace around your allure script. In your case ` ws("$workspace/target/"){  allure([..., results: [[path: "allure-results"]]]) }
`

This a complete example
```
automation_repository = 'https://github.com/fescobar/allure-docker-service.git'
default_branch = 'master'

pipeline {
    agent any
    options {
        disableConcurrentBuilds()
        buildDiscarder(logRotator(numToKeepStr: '10'))
    }

    stages {
        stage('Clone Project Testing Repository') {
            steps {
                cleanWs()
                git(url: automation_repository, branch: default_branch)
            }
        }

        stage('Run Tests') {
            steps {
                warnError('Unstable Tests') {
                    print('This stage should be use it to run tests generating allure-results directory')
                }
            }
        }

        stage('ALLURE') {
            steps {
                sh """
                ls -a ${WORKSPACE}
                """
                script {
                    ws("$workspace/allure-docker-api-usage/"){
                        allure([
                            includeProperties: false,
                            jdk: '',
                            properties: [],
                            reportBuildPolicy: 'ALWAYS',
                            results: [[path: "allure-results-example"]]
                        ])
                    }
                }
            }
        }
    }
}
```

**Comment by semihural** [comment]
@fescobar 

I got your point and tried with above but it again use below path however reports are generated below path. So when ı tried as below I received an error as below. I guess there is something missing on jenkins agent side. What I mean, ı installed allure as below but now I am confuse whether if I need to to something on agent side as well. But no sure what should I specified as HOME?

![image](https://user-images.githubusercontent.com/38153463/93202687-c5276780-f75b-11ea-823f-f5d656d1bb0f.png)


![image](https://user-images.githubusercontent.com/38153463/93202583-990be680-f75b-11ea-92e9-1c6c269b756b.png)


```
Also:   hudson.remoting.Channel$CallSiteStackTrace: Remote call to dot-playground
		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1788)
		at hudson.remoting.UserRequest$ExceptionResponse.retrieve(UserRequest.java:356)
```

                    ws("/home/ubuntu/workspace/selenium_tests@2/OneApiTestAutomation/OneApiHubTestAutomation/target/"){
                        allure([
                            includeProperties: false,
                            jdk: '',
                            properties: [],
                            reportBuildPolicy: 'ALWAYS',
                            results: [[path: 'allure-results']]
                        ])
                    }
                }
            }
        }  


```
ubuntu@play-0:~/workspace/selenium_tests@2/OneApiTestAutomation/OneApiHubTestAutomation/target/allure-results$ ls
33414778-1616-4093-90e5-fd1a25dc5eb8-attachment  57d1a403-3777-48ab-9a3f-1a740c465ea6-result.json     9a17f19b-b082-433b-9b9a-658960a742a3-container.json
473a014e-eeee-4ea6-a530-cd3fe6b27fc1-attachment  8b3453c2-76e7-4aa9-8a91-ab8b41c36ae5-container.json  d4448b2d-8496-4f94-bd1f-6cab0e0bc9cc-result.json
```

```
+ ls -a /home/ubuntu/workspace/selenium_tests
.
..
allure-report
.git
.gitignore
Jenkinsfile
OneApiTestAutomation
target
[Pipeline] script
[Pipeline] {
[Pipeline] ws
Running in /home/ubuntu/workspace/selenium_tests/OneApiTestAutomation/OneApiHubTestAutomation/target
[Pipeline] {
[Pipeline] allure
[target] $ /home/ubuntu/tools/ru.yandex.qatools.allure.jenkins.tools.AllureCommandlineInstallation/Allure_2.13.5/bin/allure generate -c -o /home/ubuntu/workspace/selenium_tests/OneApiTestAutomation/OneApiHubTestAutomation/target/allure-report
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
allure-results does not exists
Report successfully generated to /home/ubuntu/workspace/selenium_tests/OneApiTestAutomation/OneApiHubTestAutomation/target/allure-report
Allure report was successfully generated.
Creating artifact for the build.
Artifact was added to the build.
```

**Comment by semihural** [comment]
@fescobar 

I guess ı need to install allure on agent but how should I do this ? Via tools location?

```
Also:   hudson.remoting.Channel$CallSiteStackTrace: Remote call to dot-playground
		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1788)
		at hudson.remoting.UserRequest$ExceptionResponse.retrieve(UserRequest.java:356)
		at hudson.remoting.Channel.call(Channel.java:998)
		at hudson.FilePath.act(FilePath.java:1069)
		at hudson.FilePath.act(FilePath.java:1058)
		at ru.yandex.qatools.allure.jenkins.AllureReportPublisher.addTestRunInfo(AllureReportPublisher.java:399)
		at ru.yandex.qatools.allure.jenkins.AllureReportPublisher.prepareResults(AllureReportPublisher.java:390)
		at ru.yandex.qatools.allure.jenkins.AllureReportPublisher.perform(AllureReportPublisher.java:230)
		at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:80)
		at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:67)
		at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
		at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
		at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
		at java.base/java.lang.Thread.run(Thread.java:834)
java.nio.file.AccessDeniedException: /home/ubuntu/workspace/selenium_tests@2/OneApiTestAutomation/OneApiHubTestAutomation/target/allure-results/testrun.json
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
	at java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:434)
	at java.nio.file.Files.newOutputStream(Files.java:216)
	at java.nio.file.Files.newBufferedWriter(Files.java:2860)
	at ru.yandex.qatools.allure.jenkins.callables.AbstractAddInfo.invoke(AbstractAddInfo.java:26)
	at ru.yandex.qatools.allure.jenkins.callables.AbstractAddInfo.invoke(AbstractAddInfo.java:19)
	at hudson.FilePath$FileCallableWrapper.call(FilePath.java:3069)
	at hudson.remoting.UserRequest.perform(UserRequest.java:211)
	at hudson.remoting.UserRequest.perform(UserRequest.java:54)
	at hudson.remoting.Request$2.run(Request.java:369)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:72)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by fescobar** [comment]
@semihural try doing this

`ws` allows you to create a custom workspace with the name that you want. Use `ws` when you generate the results and when you generate the report. In my case I'm creating a custom workspace named `my-workspace`. I'm using `ws` in every stage.

```
automation_repository = 'https://github.com/fescobar/allure-docker-service.git'
default_branch = 'master'
my_workspace = 'my-workspace'
pipeline {
    agent any
    options {
        disableConcurrentBuilds()
        buildDiscarder(logRotator(numToKeepStr: '10'))
    }

    stages {
        stage('Clone Project Testing Repository') {
            steps {
                ws(my_workspace){
                    cleanWs()
                    git(url: automation_repository, branch: default_branch)
                }
            }
        }

        stage('Run Tests') {
            steps {
                ws(my_workspace){
                    warnError('Unstable Tests') {
                        print('This stage should be use it to run tests generating allure-results directory')
                    }
                }
            }
        }

        stage('ALLURE') {
            steps {
                ws(my_workspace+"/allure-docker-api-usage/"){
                    script {
                            allure([
                                includeProperties: false,
                                jdk: '',
                                properties: [],
                                reportBuildPolicy: 'ALWAYS',
                                results: [[path: "allure-results-example"]]
                            ])
                    }
                }
            }
        }
    }
}
```

**Comment by Vodrech** [comment]
When I created the allure report through jenkins pipeline it seemed to work correctly, but then I looked into the GUI configurations and the path were set to "allure-results" instead of the given parameter path "target/allure-results".  It seems that it doesn't take in the parameters.

1.) Must I specify the other params, or can't I just specify one?
2.) And is there something with the publisher that is causing the problem?

_Note: I tried with all the sugest parameters but it still didn't work._

reference for publisher allure:
https://docs.openstack.org/infra/jenkins-job-builder/publishers.html ( Think its from 2014 but not sure )

```
 publishers {
            allure {
                results([path:'target/allure-results'])
                }
            }
```

_Guidance:_ If someone gets NaN% in the report on jenkins, look into the results --> path and make sure its pointing to target/allure-reports and not just allure-reports.


_Note: This is just for the "General configurations, if you have set to another path/location maybe something else is causing it"._


If someone got any advice, its recommended to write those below! Still in search after a solution to auto generate the correct path for allure :(.
